src/bench/mempool_eviction.cpp

```diff
@@ -21,116 +21,65 @@ static void AddTx(const CTransactionRef& tx, const CAmount& nFee, CTxMemPool& po
-// mempool. Code needs to be written to generate a much wider variety of
-// unique transactions for a more meaningful performance measurement.
-static void MempoolEviction(benchmark::Bench& bench)
+void add_parents_child(std::vector<CTransactionRef>& txns, size_t package_size, size_t set_num)
 {
-    const auto testing_setup = MakeNoLogFileContext<const TestingSetup>();
-
-    CMutableTransaction tx1 = CMutableTransaction();
-    tx1.vin.resize(1);
-    tx1.vin[0].scriptSig = CScript() << OP_1;
-    tx1.vin[0].scriptWitness.stack.push_back({1});
-    tx1.vout.resize(1);
-    tx1.vout[0].scriptPubKey = CScript() << OP_1 << OP_EQUAL;
-    tx1.vout[0].nValue = 10 * COIN;
+    // N+1 N-input-N-output txns, in a parents-and-child package
+    // Where the final tx takes an input from all ancestors
+    const size_t num_puts = std::max(package_size - 1, (size_t)1);
+    const size_t num_txns = package_size;
+    for (size_t txn_index=0; txn_index<num_txns; txn_index++) {
+        CMutableTransaction tx = CMutableTransaction();
+        tx.vin.resize(num_puts);
+        tx.vout.resize(num_puts);
+        for (size_t put_index=0; put_index<num_puts; put_index++) {
+            tx.vin[put_index].scriptSig = CScript() << CScriptNum(txn_index);
+            tx.vin[put_index].scriptWitness.stack.push_back({1});
+            tx.vout[put_index].scriptPubKey = CScript() << CScriptNum(set_num); // Ensures unique txid with set_num
+            tx.vout[put_index].nValue = txn_index;
 
-    CMutableTransaction tx2 = CMutableTransaction();
-    tx2.vin.resize(1);
-    tx2.vin[0].scriptSig = CScript() << OP_2;
-    tx2.vin[0].scriptWitness.stack.push_back({2});
-    tx2.vout.resize(1);
-    tx2.vout[0].scriptPubKey = CScript() << OP_2 << OP_EQUAL;
-    tx2.vout[0].nValue = 10 * COIN;
-
-    CMutableTransaction tx3 = CMutableTransaction();
-    tx3.vin.resize(1);
-    tx3.vin[0].prevout = COutPoint(tx2.GetHash(), 0);
-    tx3.vin[0].scriptSig = CScript() << OP_2;
-    tx3.vin[0].scriptWitness.stack.push_back({3});
-    tx3.vout.resize(1);
-    tx3.vout[0].scriptPubKey = CScript() << OP_3 << OP_EQUAL;
-    tx3.vout[0].nValue = 10 * COIN;
-
-    CMutableTransaction tx4 = CMutableTransaction();
-    tx4.vin.resize(2);
-    tx4.vin[0].prevout.SetNull();
-    tx4.vin[0].scriptSig = CScript() << OP_4;
-    tx4.vin[0].scriptWitness.stack.push_back({4});
-    tx4.vin[1].prevout.SetNull();
-    tx4.vin[1].scriptSig = CScript() << OP_4;
-    tx4.vin[1].scriptWitness.stack.push_back({4});
-    tx4.vout.resize(2);
-    tx4.vout[0].scriptPubKey = CScript() << OP_4 << OP_EQUAL;
-    tx4.vout[0].nValue = 10 * COIN;
-    tx4.vout[1].scriptPubKey = CScript() << OP_4 << OP_EQUAL;
-    tx4.vout[1].nValue = 10 * COIN;
+            if (put_index < txns.size()) {
+                // We spend put_index's txn's in the next available slot for each previous transaction
+                tx.vin[put_index].prevout = COutPoint(txns[put_index]->GetHash(), put_index + txn_index - 1);
```

Comment:
```
bench/mempool_eviction.cpp:42:105: runtime error: unsigned integer overflow: 0 - 1 cannot be represented in type 'unsigned long'

Reply:
you running this without any args? txn_index has to be greater than 0, so unsure how that would happen

Reply:
found the issue, thanks